exporter/clickhouselogsexporter/logsv2/fingerprint.go

```diff
@@ -0,0 +1,195 @@
+package logsv2
+
+import (
+	"fmt"
+	"strings"
+)
+
+type DimensionHierarchyNode struct {
+	// labels that map to this node in dimension hierarchy
+	// TODO(Raj): Can labels be a concat of 2 keys?
+	labels []string
+
+	// List of potential subhierachies in the order of preference.
+	// Eg: k8s.cluster.name can have subhierarchies of k8s.namespace.name or k8s.node.name - the 2 ways of grouping/organizing k8s resources.
+	// In most cases this list will have only one entry
+	subHierachies []DimensionHierarchyNode
+}
+
+type IdLabelValue struct {
+	Label string
+	Value any
+}
+
+// Returns list of dimension labels for a set of attributes for a DimensionHierarchy
+func (node *DimensionHierarchyNode) Identifier(attributes map[string]any) []IdLabelValue {
+	result := []IdLabelValue{}
+
+	for _, l := range node.labels {
+		if lVal, exists := attributes[l]; exists {
+			result = append(result, IdLabelValue{
+				Label: l,
+				Value: lVal,
+			})
+			break
+		}
+	}
+
+	for _, s := range node.subHierachies {
+		subLabels := s.Identifier(attributes)
+		if len(subLabels) > 0 {
+			result = append(result, subLabels...)
+			break
+		}
+	}
+
+	return result
+}
+
+// TODO(Raj/Nitya): Consider parsing this stuff out from json
+func ResourceHierarchy() *DimensionHierarchyNode {
+	return &DimensionHierarchyNode{
+		labels: []string{
+			"cloud.provider",
+			"source_type",
```

Comment:
so `source_type` is for vector logs. It is by default added by vector.

eg:- `k8s`, `file` . So it's kind of top-level for vector

https://signoz.io/docs/logs-management/send-logs/vector-logs-to-signoz/#mapping-for-signoz

Reply:
If we put as a synonym for `cloud.platform` then it will be a child of `gcp.project` . What we saw was that `source_type` was already present for all and some of them also had `gcp.project` . 

Do you see any issue with that?